pkg/controller/warehouses/warehouses.go

```diff
@@ -259,6 +262,7 @@ func (r *reconciler) syncWarehouse(
 					ObservedGeneration: warehouse.GetGeneration(),
 				},
 			)
+			conditions.Delete(&status, kargoapi.ConditionTypeCriteriaSatisfied)
```

Comment:
imho, this is misplaced. As it stands, we're clearing this condition only if there's been an error during discovery. I think it would make more sense to clear it up front at the same time we initially set Reconciling to true. This way, the CriteriaSatisfied condition remains absent throughout discovery.

cc @hiddeco for some confirmation of this.